test/test_reductions.py

```diff
@@ -467,12 +467,24 @@ def test_logsumexp(self, device):
         expected = logsumexp(a.cpu().numpy(), 1)
         self.assertEqual(expected.shape, actual.shape)
         self.assertEqual(expected, actual)
+
         # check that out is actually inplace
         b = torch.zeros(5, 2, device=device)
         c = b[:, 0]
         torch.logsumexp(a, 1, out=c)
         self.assertEqual(expected, b[:, 0])
 
+        # check integral inputs is promoted to floating point
+        e = torch.randint(-100, 100, [5, 4], device=device)
+        actual = e.logsumexp(1).to(torch.float64)
+        expected = logsumexp(e.cpu().numpy(), 1)
+        self.assertEqual(expected.shape, actual.shape)
+        self.assertEqual(expected, actual)
+
+        for dtype in [torch.bool, torch.int8, torch.uint8, torch.int16, torch.int32, torch.int64]:
```

Comment:
I think this check is covered here:

https://github.com/pytorch/pytorch/blob/ea0de37d2e5aa71178580f3bce7559cc6a3b9135/test/test_ops.py#L356

That check is not as thorough as what you've written but I think it should be sufficient.